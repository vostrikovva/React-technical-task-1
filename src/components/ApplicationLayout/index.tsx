import { FC } from 'react'

import styles from './styles.module.css'

interface ApplicationLayoutProps {
  children?: React.ReactNode
}

const ApplicationLayout: FC<ApplicationLayoutProps> = ({ children }) => {
  return (
    <>
        <div className='p-3'>
            <h1>
                Doctor booking service
            </h1>
            <hr />
        </div>
        <div className={styles.content}>
            {children}
        </div>
    </>
  )
}

export default ApplicationLayout
