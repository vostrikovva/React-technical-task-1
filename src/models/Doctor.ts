import Address from './Address'

export default interface Doctor {
  id: string
  name: string
  address?: Address
}

export interface BookDoctorBody {
  doctorId: string
  date: string
}

export interface BookDoctorResponse {
  doctorId: string
  date: string
  createdAt: string
  updatedAt: string
  id: string
}
