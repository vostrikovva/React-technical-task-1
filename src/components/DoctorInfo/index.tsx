import { FC, memo, useCallback } from 'react'
import { Card, Button } from 'react-bootstrap'
import clsx from 'clsx'

import styles from './styles.module.css'
import Doctor from '../../models/Doctor'

interface DoctorInfoProps {
  doctor: Doctor
  setBookedDoctor: (doctor: Doctor | null) => void
}

const DoctorInfo: FC<DoctorInfoProps> = ({ doctor, setBookedDoctor }) => {
  const { name, address } = doctor

  const handleOnClick = useCallback(() => {
    setBookedDoctor(doctor)
  }, [setBookedDoctor, doctor])

  return (
    <Card className={clsx(styles.root, 'm-2')}>
      <Card.Body>
        <Card.Title className="mb-3">{name}</Card.Title>
        {address &&
          <div className="row mb-10">
            <div className="col-auto d-flex align-items-center" >
              <Card.Subtitle>Address</Card.Subtitle>
            </div>
            <div className="col">
              <Card.Text className="mb-0">{`${address.country}, ${address.city}`}</Card.Text>
              <Card.Text className="mb-0">Postal Code: {address.postalCode}</Card.Text>
              <Card.Text className="mb-0">Address line 1: {address.line1}</Card.Text>
              <Card.Text className="mb-0">Address line 2: {address.line2}</Card.Text>
            </div>
          </div>
        }
        <div className="mt-2 d-flex justify-content-end">
            <Button variant='primary' onClick={handleOnClick}>Booking</Button>
        </div>
      </Card.Body>
    </Card>
  )
}

export default memo(DoctorInfo)
