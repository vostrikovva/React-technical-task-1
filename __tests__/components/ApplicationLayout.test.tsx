import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

import ApplicationLayout from '../../src/components/ApplicationLayout'
import DoctorInfo from '../../src/components/DoctorInfo'
import Doctor from '../../src/models/Doctor'

describe('ApplicationLayout component', () => {
  afterEach(() => {
    cleanup()
  })

  test('Render the component without children', async () => {
    const component = render(<ApplicationLayout />)

    expect(await component.findByText('Doctor booking service')).toBeInTheDocument()
  })

  test('Render the component with DoctorInfo', async () => {
    const mockFunction = jest.fn(() => {})

    const doctor: Doctor = {
      id: 'some-id-here',
      name: 'John Snow'
    }

    const component = render(<ApplicationLayout>
        <DoctorInfo doctor={doctor} setBookedDoctor={mockFunction} />
    </ApplicationLayout>)

    expect(await component.findByText('Doctor booking service')).toBeInTheDocument()
    expect(await component.findByText(doctor.name)).toBeInTheDocument()
    expect(mockFunction).not.toHaveBeenCalled()
  })

  test('Snapshot test', () => {
    const tree = renderer.create(<ApplicationLayout />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
