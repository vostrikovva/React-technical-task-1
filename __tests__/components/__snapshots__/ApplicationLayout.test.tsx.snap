// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationLayout component Snapshot test 1`] = `
[
  <div
    className="p-3"
  >
    <h1>
      Doctor booking service
    </h1>
    <hr />
  </div>,
  <div
    className="content"
  />,
]
`;
